src/main/java/com/aphyr/riemann/client/ChainPromise.java

```diff
@@ -92,6 +92,11 @@ public Object unsafeDeref(final long time,
   }
 
   @Override
+  public void setCallback(Callback<T> callback) {
+    throw new UnsupportedOperationException("Can't set a callback on a chained promise; set it on the underlying promise instead.");
```

Comment:
Gonna need it in order for bulkClient's promises to be useful. Gotta set a callback on the promise here, which in turn sets a callback on *that* promise.